### ISSUE TITLE
Menu and ViewController index mismatch when trying to update menu title using api calls

### ISSUE THREAD

**Issue by davidhsu1115** [rechsteiner/Parchment#588]
As Title, I am fetching title from api service, however once I update the menu at second page of the view controller, the menu index and viewcontroller index cannot be matched.
It will auto select first menu tab. But it should be at second one.

**Comment by davidhsu1115** [comment]
Update: if I reloadData, the there isn't any issue at all, but it will always pop to first page once every menu update.

**Comment by davidhsu1115** [comment]
After some digging in closed and open issue, I think the cause of this issue might be the changing title cant match the hashing, that's way after reloadMenu cause mismatch index. Is there any way or example to solve this kind of problem?
Thx.

**Comment by tonylam1211** [comment]
@davidhsu1115 hello~~ I met the same problem with you. After reloadMenu, mismatching index happened. Do you still remember how did you solve this?? Thank you so much!!

**Comment by rechsteiner** [comment]
Hi @tonylam1211! Could you check out the 4.0 branch and see if the issue is still present?

**Comment by tonylam1211** [comment]
Hi! Yes, the issue still exists. 
=========================
My case is that, I would like to do the following:
I had SOME NEW menu item with some special style. After the user tapped the NEW item, it would change back to normal style. ("tapped": not the selecting one, but selected && jumped to another menu item)

So, in my custom PagingItem, I added a variable like 
`var isNew: Bool`

`func pagingViewController(_: PagingViewController, pagingItemAt index: Int) -> PagingItem {
......
// Some DS to update if user clicked the menu item or not
// isNew: true --> new special item and NEVER tapped
// isNew: false --> normal item || new special item but tapped
return CustomItem(..., isNew: isNew)
}`

And after I called the reloadMenu in didSelectItem/didScrollToItems, things go normal for normal items.
However, when it comes to new special item, the menu always jumps to the first item.
========================

<img width="1286" alt="Screenshot 2023-05-03 at 9 23 03 AM" src="https://user-images.githubusercontent.com/49671376/235817430-55c8fbbe-4a58-4961-8d80-85ddeadaed27.png">

========================

And I think the reason is that, 1st red box updated the dataSource, which causes the "equal" statement in 2nd red box not triggered:

before: CustomItem(..., isNew: true)
updated: CustomItem(..., isNew: false)

which makes the menu jumps into the else if case and goes to the first item

========================

Any suggestions on how should this be done? Thank you!

**Comment by davidhsu1115** [comment]
I don't have the project currently, but in my memory I might have implemented my own custom PagingItem to use only index as comparing condition to solve this.

**Comment by tonylam1211** [comment]
@davidhsu1115 Thank you for your advice!! I tried to make use of index on PagingItem but still having some troubles solving the issue thoroughly...
------
I made some changes on the PagingViewController which seems okay to make it work in my project. But as I am not too familiar with Parchment, I am not sure if there is/are any side-effects to do so. 

@rechsteiner May you help fix the issue? Thank you so much!!
------
My changes are as followed:
<img width="1291" alt="Screenshot 2023-05-03 at 5 53 08 PM" src="https://user-images.githubusercontent.com/49671376/236084902-ae679797-6246-4ca9-b788-a2489651cdf8.png">
------
